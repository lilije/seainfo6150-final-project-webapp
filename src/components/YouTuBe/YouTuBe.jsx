import React from 'react';
import styles from './YouTuBe.module.css';

class YouTuBe extends React.Component {
  
  displayPosts = () => {
    const listItems = this.props.videos.map((video, index) =>
      <iframe height="315" title="1" src={video} frameBorder="0" key={index}
        allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowFullScreen>
        </iframe>);
    return (listItems);
  }
    
  
  render() {
    return (
      <div className={styles.content}>
        {this.displayPosts()}
      </div>
    )
  }
}

export default YouTuBe;