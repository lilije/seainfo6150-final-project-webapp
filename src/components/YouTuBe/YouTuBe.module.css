.content {
  display: flex;
  padding-top: 100px;
  margin: 0 auto;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  padding-bottom: 120px;
  max-width: 1500px;
}

iframe {
  width: 100%;
  margin: 20px;
}

@media screen and (min-width: 1200px) {
  .content {
    flex-direction: row;
    flex-wrap: wrap;
  }
  
  iframe {
    width: 30%;
  }
}