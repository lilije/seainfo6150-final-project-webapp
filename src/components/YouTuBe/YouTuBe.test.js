import React from "react";
import { render } from "@testing-library/react";
import YouTuBe from "./YouTuBe.jsx";

it("renders the YouTuBe component correctly", () => {
  const videos = ["https://www.youtube.com/embed/2AuPF74q-fY"];
  const { container } = render(<YouTuBe videos={videos}/>);
  expect(container).toMatchSnapshot();
});