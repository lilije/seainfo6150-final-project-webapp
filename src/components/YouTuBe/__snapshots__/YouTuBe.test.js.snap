// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the YouTuBe component correctly 1`] = `
<div>
  <div
    class="content"
  >
    <iframe
      allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
      allowfullscreen=""
      frameborder="0"
      height="315"
      src="https://www.youtube.com/embed/2AuPF74q-fY"
      title="1"
    />
  </div>
</div>
`;
